File: src/Cxx/Images/ImageSinusoidSource.cxx
#include <vtkSmartPointer.h>
#include <vtkImageViewer2.h>
#include <vtkImageSinusoidSource.h>
#include <vtkRenderWindow.h>
#include <vtkRenderWindowInteractor.h>
#include <vtkRenderer.h>

int main(int, char*[])
{
  vtkSmartPointer<vtkImageSinusoidSource> sinusoidSource =
    vtkSmartPointer<vtkImageSinusoidSource>::New();
  sinusoidSource->Update();

  // Visualize
  vtkSmartPointer<vtkImageViewer2> imageViewer =
    vtkSmartPointer<vtkImageViewer2>::New();
  imageViewer->SetInputConnection(sinusoidSource->GetOutputPort());
  imageViewer->GetRenderWindow()->SetSize( 500, 500 );
  imageViewer->GetRenderer()->ResetCamera();

  // Set up an interactor that does not respond to mouse events
  vtkSmartPointer<vtkRenderWindowInteractor> renderWindowInteractor =
    vtkSmartPointer<vtkRenderWindowInteractor>::New();
  imageViewer->GetRenderWindow()->SetInteractor( renderWindowInteractor );
  renderWindowInteractor->SetInteractorStyle( 0 );
  imageViewer->Render();

  // Start the event loop
  renderWindowInteractor->Initialize();
  renderWindowInteractor->Start();

  return EXIT_SUCCESS;
}

File: src/Java/Images/ImageSinusoidSource.java
import vtk.vtkNativeLibrary;
import vtk.vtkImageSinusoidSource;
import vtk.vtkImageViewer2;
import vtk.vtkRenderWindowInteractor;

public class ImageSinusoidSource 
{
  // -----------------------------------------------------------------
  // Load VTK library and print which library was not properly loaded
  static 
  {
    if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
    {
      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
      {
        if (!lib.IsLoaded()) 
        {
          System.out.println(lib.GetLibraryName() + " not loaded");
        }
      }
    }
    vtkNativeLibrary.DisableOutputWindow(null);
  }
  // -----------------------------------------------------------------

public static void main(String args[]) 
{
  
  vtkImageSinusoidSource sinusoidSource = new vtkImageSinusoidSource();
  sinusoidSource.Update();

  //Visualize
  vtkImageViewer2 imageViewer = new vtkImageViewer2();
  imageViewer.SetInputConnection(sinusoidSource.GetOutputPort());
  imageViewer.GetRenderWindow().SetSize( 500, 500 );
  imageViewer.GetRenderer().ResetCamera();

  vtkRenderWindowInteractor iren = new vtkRenderWindowInteractor();
  imageViewer.GetRenderWindow().SetInteractor(iren);
  iren.SetInteractorStyle(null);
  imageViewer.Render();

  iren.Initialize();
  iren.Start();

 }
}
